cmd/juju/application/storage.go

```diff
@@ -0,0 +1,333 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package application
+
+import (
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/juju/cmd/v3"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"github.com/juju/names/v5"
+	"github.com/juju/utils/v3"
+
+	"github.com/juju/juju/api/client/application"
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/juju/config"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/cmd/output"
+	"github.com/juju/juju/storage"
+)
+
+const (
+	appStorageConfigSummary = "Displays or sets storage directives for an application."
+
+	appStorageConfigDoc = `
+To view all storage directives for the given application:
+
+    juju application-storage <application>
+
+	By default, the config will be printed in a tabular format. You can instead
+print it in ` + "`json`" + ` or ` + "`yaml`" + ` format using the ` + "`--format`" + ` flag:
+
+   	juju application-storage <application> --format json
+    juju application-storage <application> --format yaml
+
+To view the directive of a single storage name:
+
+    juju application-storage <application> <storage-name>
+
+To set storage directives on an application:
+
+    juju application-storage <application> name1=size,pool,count name2=size,pool,count
+
+Config values can be imported from a yaml file using the ` + "`--file`" + ` flag:
+
+    juju application-storage <application> --file=path/to/config.yaml
+
+This allows you to, e.g., save an application's storage directives to a file:
+
+    juju application-storage <application> --format=yaml > config.yaml
+
+and then import the config later. 
```

Comment:
Sorry, I missed this before. I don't think we need this. The text has been copied from the juju config command and tweaked so doesn't quite read correctly. But regardless, allowing storage to be defined in a file is not something we currently support for deploy or refresh, so should not be supported here either. Unlike config where there can be many, many options, it's a few at the most. But that aside, we either support it for both deploy and yhis new command, or not at all.

Reply:
Sure will remove the functionality for passing configs from file as well